tests/FSharp.Configuration.Tests/YamlProvider.Tests.fs

```diff
@@ -176,7 +176,38 @@ TopLevelUnknown:
 
 """
     settings.Mail.Smtp.Port |> should equal 4430
-    
+
+
+[<Test>]
+let ``Can load file and watch``() =
+    let settings = Settings()
+    let tempFile = Path.GetTempFileName()
+    try
+      File.Copy ("Settings.yaml", tempFile, overwrite=true)
+      settings.LoadAndWatch tempFile |> ignore
+    finally
+      File.Delete tempFile
+
+[<Test>]
+let ``Can load file and watch and detect errors``() =
+    let settings = Settings()
+    let tempFile = Path.GetTempFileName()
+    let mutable err = false
+    settings.Error.Add(fun _ -> err <- true) 
+    try
+      File.Copy ("Settings.yaml", tempFile, overwrite=true)
+      use x = settings.LoadAndWatch tempFile
+      System.Threading.Thread.Sleep(800)
+      let f = new FileStream(tempFile, FileMode.Append)
+      let data = "<junk:>:asd:DF" |> System.Text.Encoding.ASCII.GetBytes
+      f.Write(data, 0, data.Length)
+      f.Flush()
+      f.Dispose()
+      System.Threading.Thread.Sleep(800)
```

Comment:
Partially, yes.  The first sleep is required because the internal file watcher code won't pass through file system changes that happened less than 500 ms apart.  So I have to wait at least 500 ms after the LoadAndWatch call or I won't be notified of the changes. I could change the second sleep to a timed-out event wait that was set in the event handler, perhaps.  I agree the sleeps are nasty, but I can only get rid of one of them easily.